### ISSUE TITLE
Test failure caused by #556/T710-008

### ISSUE THREAD

**Issue by LordAro** [AdaCore/libadalang#579]
Caused by the memoisation change in #556 

```
INFO     FAIL            navigation__subp_top_level: unexpected output
Diff failure: unexpected output
--- expected
+++ output
@@ -8,4 +8,3 @@
-Body previous part of <SubpBody ["Foo"] foo.adb:1:1-7:9> is <SubpDecl ["Foo"] foo.ads:3:1-3:25> [foo.ads]
-Decl of <SubpBody ["Foo"] foo.adb:1:1-7:9> is <SubpDecl ["Foo"] foo.ads:3:1-3:25> [foo.ads]
+Error when processing <SubpBody ["Foo"] foo.adb:1:1-7:9>
```

```
INFO     FAIL            checks__stylechecks: unexpected output
Diff failure: unexpected output
--- expected
+++ output
@@ -0,0 +1,2 @@
+ada/language/ast.py:2413: Too long line
+ada/language/ast.py:2413:79: E501 line too long (81 > 79 characters)
```
(Why are the stylechecks part of the unit tests rather than an individual check? Would make them easier to spot before merging)

**Comment by raph-amiard** [comment]
> (Why are the stylechecks part of the unit tests rather than an individual check? Would make them easier to spot before merging)

Hi @LordAro ,

It's so our most uncompliant devs don't forget to run them :) Since running the testsuite is a prereq. Clearly didn't work out there. Also, what do you mean by "easier to spot" ?

Also, thanks for the report! We're on it :)

**Comment by LordAro** [comment]
Cheers, I figured you'd probably noticed it already, but I thought I might as well make sure of it :)

By "easier to spot" I meant having a separate GH Action for it, then the style check failure would show up separately to test failures. But of course, I've just looked and you're only using GH Actions for MacOS build. Something to consider in future if you ever fully move to GH Aactions, perhaps :)

**Comment by raph-amiard** [comment]
> By "easier to spot" I meant having a separate GH Action for it, then the style check failure would show up separately to test failures. But of course, I've just looked and you're only using GH Actions for MacOS build. Something to consider in future if you ever fully move to GH Aactions, perhaps :)

That's a great comment indeed. We plan to migrate more of our public CIs to GH actions, and we'll probably consider this! Thanks a lot :slightly_smiling_face:

**Comment by LordAro** [comment]
Specific failures fixed by #580 & #581 ... but dsl_unparse is broken due to a langkit change, so I won't close the issue yet :p

**Comment by pmderodat** [comment]
That last testcase will be fixed with #584, so we should be good after that. ;-)
